ironic_prometheus_exporter/parsers/ipmi.py

```diff
@@ -0,0 +1,306 @@
+import re
+
+# NOTE (iurygregory): most of the sensor readings come in the ipmi format
+# each type of sensor consider a different range of values that aren't integers
+# (eg: 0h, 2eh), 0h will be published as 0 and the other values as 1, this way
+# we will be able to create prometheus alerts.
+# Documentation: https://www.intel.com/content/www/us/en/servers/ipmi/
+# ipmi-second-gen-interface-spec-v2-rev1-1.html
+
+
+def add_prometheus_type(name, metric_type):
+    return '# TYPE %s %s' % (name, metric_type)
+
+
+def metric_names(payload, prefix, sufix, extract_unit=False,
+                 special_label=None):
+    metric_dic = {}
+    for entry in payload:
+        if special_label == 'fan':
+            e = re.sub(r'[\d].*$', '', entry.lower())
+            e = re.sub(r'[\(\)]', '', e).split()
+            label = '_'.join(e)
+        else:
+            e = re.sub(r"[\d]+", "", entry).lower().split()
+            label = '_'.join(e[:-1]).replace('-', '_')
+
+        if extract_unit:
+            sensor_read = payload[entry]['Sensor Reading'].split()
+            if len(sensor_read) > 1:
+                sufix = '_' + sensor_read[-1].lower()
+
+        if special_label == 'memory':
+            if 'mem' not in label and 'memory' not in label:
+                label = 'memory_' + label
+
+        metric_name = prefix + label + sufix
+        if metric_name in metric_dic:
+            metric_dic[metric_name].append(entry)
+        else:
+            metric_dic[metric_name] = [entry]
+    return metric_dic
+
+
+def extract_labels(entries, payload, node_name):
+    default_label = 'node_name="%s"' % node_name
+    if len(entries) == 1:
```

Comment:
Why does a len of 1 mean there is no sensor label?

Reply:
by sensor label you mean "Sensor ID" ?  (Each 
I only added the Sensor ID information as label when the metric name matches for more than 1 Sensor ID (eg. in Temperature we have "Temp (0x1)" and "Temp (0x2)" since the metric name for them is the same the Sensor ID information is added as label.
baremetal_temperature_celcius{node_name=...,sensor=Temp1

The len 1 means that the metric only comes from one sensor so I  only need to add the node_name as label.

Reply:
Ah, ok, that makes sense. Some comments about this sort of thing (and sample data) would make it easier to understand the logic throughout.

Reply:
I will add